src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -335,6 +374,78 @@ final class HTTPClient: Client {
 			}
 		}
 	}
+
+	private func otpAuthorizationSuccessHandler(
+		for response: URLSession.Response,
+		completion: @escaping OTPAuthorizationCompletionHandler
+	) {
+		guard let responseBody = response.body else {
+			Log.error("Failed to authorize OTP - response error", log: .api)
+			Log.error(String(response.statusCode), log: .api)
+			completion(.failure(.invalidResponseError))
+			return
+		}
+		do {
+			let decoder = JSONDecoder()
+			decoder.dateDecodingStrategy = .iso8601
+			let decodedResponse = try decoder.decode(
+				OTPResponseProperties.self,
+				from: responseBody
+			)
+			guard let expirationDate = decodedResponse.expirationDate else {
+				Log.error("Failed to get expirationDate out of decoded response", log: .api)
+				completion(.failure(.invalidResponseError))
+				return
+			}
+			completion(.success(expirationDate))
+		} catch {
+			Log.error("Failed to get expirationDate because of invalid response payload structure", log: .api)
+			completion(.failure(.invalidResponseError))
+		}
+	}
+
+	private func otpAuthorizationFailureHandler(
+		for response: URLSession.Response,
+		completion: @escaping OTPAuthorizationCompletionHandler
+	) {
+		guard let responseBody = response.body else {
+			Log.error("Failed to get authorized OTP - no 200 status code", log: .api)
+			Log.error(String(response.statusCode), log: .api)
+			completion(.failure(.invalidResponseError))
+			return
+		}
+		do {
+			let decodedResponse = try JSONDecoder().decode(
+				OTPResponseProperties.self,
+				from: responseBody
+			)
+			guard let errorCode = decodedResponse.errorCode else {
+				Log.error("Failed to get errorCode because of invalid response payload structure", log: .api)
+				completion(.failure(.invalidResponseError))
+				return
+			}
+
+			switch errorCode {
+			case .API_TOKEN_ALREADY_ISSUED:
+				completion(.failure(.apiTokenAlreadyIssued))
+			case .API_TOKEN_EXPIRED:
+				completion(.failure(.apiTokenExpired))
+			case .API_TOKEN_QUOTA_EXCEEDED:
+				completion(.failure(.apiTokenQuotaExceeded))
+			case .DEVICE_TOKEN_INVALID:
+				completion(.failure(.deviceTokenInvalid))
+			case .DEVICE_TOKEN_REDEEMED:
+				completion(.failure(.deviceTokenRedeemed))
+			case .DEVICE_TOKEN_SYNTAX_ERROR:
+				completion(.failure(.deviceTokenSyntaxError))
+			default:
+				completion(.failure(.otherServerError))
+			}
+		} catch _ {
+			Log.error("Failed to get errorCode because json could not be decoded", log: .api)
```

Comment:
You can pass the generic error to the log:
```suggestion
		} catch {
			Log.error("Failed to get errorCode because json could not be decoded", log: .api, error: error)
```